src/components/MovieLibrary.jsx

```diff
@@ -1 +1,83 @@
-// implement MovieLibrary component here
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import MovieList from './MovieList';
+import SearchBar from './SearchBar';
+import AddMovie from './AddMovie';
+
+class MovieLibrary extends Component {
+  constructor(props) {
+    super(props);
+    const { movies } = this.props;
+    this.state = {
+      searchText: '',
+      bookmarkedOnly: false,
+      selectedGenre: '',
+      movies,
+    };
+  }
+
+  // v feito com ajuda do Matheus Duarte
+
+  handleChange = (e) => {
+    const { name } = e.target;
+    let value = '';
+    if (e.target.type === 'checkbox') {
+      value = e.target.checked;
+    } else {
+      value = e.target.value;
+    }
```

Comment:
Acredito que vai funcionar também desse jeito.
```suggestion
    const { name, type, checked } = e.target;
    let { value } = e.target;
    if (type === 'checkbox') {
      value = checked;
    }
```